### PR TITLE
New notifications

### PR DESCRIPTION
**Pull request by eoji**
# what
Notification settings redesign.

# why
We have new designs, THAT ARE AWESOME.

# how
100% kotlin 😎 
I used ✨ Kotlin Android extensions ✨ to handle the click listeners.
I used our existing `ViewModel` pattern and it was iconic.
I added new icons for the phone icon.
I added new drawables for the icon backgrounds.
I added a new circular ripple drawable for the icon foreground on 21+.
I wrote tests to handle toggling all of the notification settings.
I wrote an instrumentation test to make sure the `ProjectNotificationSettingsActivity` starts when users click the `Manage project notifications` view.
In conclusion, this is my cutest PR.

*FYI*: Going forward, Danny said we'll be using the standard `16dp` margin on the `start` and `end` of full width elements so I added that value in.

# see

### CODE REVIEW

**Comment by eoji** [comment]
[Re-factored notification settings - Android](https://trello.com/c/VbmlSfvK/821-re-factored-notification-settings-android)

File: app/src/main/res/drawable-v21/circular_click_indicator_light.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+  android:color="@color/ripple_light">
+  <item android:id="@android:id/mask">
+    <shape android:shape="oval">
+      <solid android:color="?android:colorPrimary"/>
+    </shape>
+    <color android:color="@color/white"/>
+  </item>
+</ripple>
```
Comment (Rcureton) [inline app/src/main/res/drawable-v21/circular_click_indicator_light.xml]: NL

File: app/src/main/java/com/kickstarter/viewmodels/NotificationsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            return this.userOutput
+        }
+
+        override fun unableToSavePreferenceError(): Observable<String> {
+            return this.unableToSavePreferenceError
+                    .takeUntil(this.updateSuccess)
+                    .map { _ -> null }
+        }
+    }
+}
```
Comment (Rcureton) [inline app/src/main/java/com/kickstarter/viewmodels/NotificationsViewModel.kt]: NL
